basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/ListHeadTailNodeType.java

```diff
@@ -0,0 +1,66 @@
+package club.mondaylunch.gatos.basicnodes;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.data.ListDataType;
+import club.mondaylunch.gatos.core.data.OptionalDataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class ListHeadTailNodeType extends NodeType.Process {
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of(
+            "head_mode", DataType.BOOLEAN.create(true)
+        );
+    }
+
+    @Override
+    public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> state) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "input", ListDataType.GENERIC_LIST),
+            new NodeConnector.Input<>(nodeId, "in", OptionalDataType.GENERIC_OPTIONAL)
```

Comment:
What's the optional for? And why are there two inputs with such similar names?

Reply:
I don't think I was thinking when I put this here. It's gone now.